### ISSUE TITLE
Sometimes eslint passes, sometimes it doesn't

### ISSUE THREAD

**Issue by cullylarson** [schoero/eslint-plugin-readable-tailwind#341]
### Description

I'm encountering this issue where, when I run a lint check, sometimes it passes and sometimes it finds issues. I even wrote a script to run a bunch of lint checks over and over on the same code, and it really does seem to be nondeterministic.

This is what Claude seems to think the root problem is:

> eslint-plugin-better-tailwindcss uses synckit worker threads to bridge Tailwind CSS's async API into synchronous ESLint rules. When ESLint processes many files, the worker encounters a race condition that causes it to silently fail. This failure corrupts the entire ESLint run — not just better-tailwindcss rules but ALL rules return 0 results. The behavior is non-deterministic: the same command run multiple times gives different results.

The issue seems to be related to the number of files being checked. When I run it on fewer files, it catches the issues almost every time. For example, let's say I have 200 files to lint, if I lint them in batches of 10 files, it works (reports the lint issues). If I lint all 200, it sometimes works, sometimes fails silently (i.e. looks like a successful run, without issues).

Sorry I don't have any more information. If you have some test projects with lots of files, it might be worth seeing if it's an issue there too.

### Flavor

TSX

### Code Input

```tsx
There isn't specific code that is causing this issue
```

### Expected Behavior

```tsx
NA
```

### Actual Behavior

```tsx
NA
```

### ESLint Log

```shell
There is no output when this bug is triggered
```

### ESLint Config

```typescript
Doesn't seem to be relevant. The lint rules do work, just not all the time.
```

### Versions

- ESLint: 9.39.2
- Parser: @typescript-eslint/parser 8.55.0
- Plugin: 4.3.0
- Node: 22.22.0

**Comment by schoero** [comment]
Does this happen locally via CLI, in the Editor or in CI environments?

I'm asking because this happens from time to time in GitHub actions where the synckit workers just hang forever and eventually timeout.

However, this should not silently fail, instead it throws an error. 

I spent some time debugging this but never found a solution unfortunately.
